### ISSUE TITLE
Remove <customTypes/>

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#5751]
After deprecation in 3.10 https://github.com/jOOQ/jOOQ/issues/5750

**Comment by bnordli** [comment]
I still see the need for `customTypes`, we are in fact using it in for the following case:

We are defining several databases (or it could just be different apis over the same database) in distinct projects under one parent maven project. The parent pom defines the general handling of common classes like this:
```
<customTypes>
  <customType>
    <name>Boolean</name>
    <type>java.lang.Boolean</type>
    <converter>our.BooleanConverter</converter>
  </customType>
  <customType>
    <name>ZonedDateTime</name>
    <type>java.time.ZonedDateTime</type>
    <converter>our.ZonedDateTimeConverter</converter>
  </customType>
  ...
</customTypes>
```
Each database/api project then references these custom type like this:
```
<forcedTypes>
  <forcedType>
    <name>Boolean</name>
    <expression>is_.*</expression> // <-- here we potentially add more expressions per project
    <types>tinyint</types>
  </forcedType>
  <forcedType>
    <name>ZonedDateTime</name>
    <types>timestamp</types>
  </forcedType>
  ...
</forcedTypes>
```
With this, we don't need to reference the converters in every database project.
Do you have a suggestion on how to do this without `customTypes`, or do you still think `customTypes` has merit?
(I agree that the ZonedDateTime type can simply be moved to the parent pom, since it most probably applies to all `timestamp`s, but the Boolean type applies differently in each project.)

**Comment by lukaseder** [comment]
Thanks for your feedback, @bnordli. I understand that this may be valuable for some cases where there are lots of forced type configurations. The benefit is replacing the repetition of 2 XML elements by 1.

In most cases, by carefully choosing unions in regular expression (e.g. `is_.*|.*_deleted|.*_active`), it is already possible to reduce the overhead of defining such types.

The reason for the deprecation is the fact that the maintenance of the code responsible for this particular feature, and keeping it backwards compatible, is rather high and has been quite error prone in the past. It would be good if we could remove this from the code generator.

**Comment by bnordli** [comment]
Thanks for your quick response!

I understand that this is a somewhat small use case, I just wanted to point out that the note in #5750: "the customTypes configuration isn't really needed any longer" isn't 100% correct.

But as long as you are aware of this case, and weigh it against general code health, I will certainly accept your decision, as we will find a way to work around its removal.

**Comment by lukaseder** [comment]
Yeah, I'm aware of your use case :). Perhaps, there should be some weight on the word *"really"* in that sentence:

> "the customTypes configuration isn't *really* needed any longer"

In any case, version 4.0 is still some time ahead in the future. Until then, the feature will be continued to be supported.

**Comment by bnordli** [comment]
That's great then, no worries!

I just saw a depreciation warning when testing the new version, and wanted to chime in.
